File: Programmers/pro_12981_영어끝말잇기.py
from collections import defaultdict
def solution(n, words):
    word_dict = defaultdict(int)
    for i in range(len(words)):
        word_dict[words[i]]+=1
        if(i>0 and words[i-1][-1] != words[i][0] or word_dict[words[i]]>1):
            return [i%n+1,i//n+1]
    return [0,0]

File: Programmers/pro_12981_영어 끝말잇기.js
function solution(n, words) {
    const dict_word = []
    dict_word[words[0]] = 1
    for (let i = 1; i < words.length; i++) {
        let [prev, cur] = [words[i - 1], words[i]]
        let [last, first] = [prev[prev.length - 1], cur[0]]
        if (dict_word[cur] || last != first)
            return [parseInt(i % n) + 1, parseInt(i / n) + 1]
        dict_word[cur] = 1
    }
    return [0, 0];
}